### PR TITLE
Death to swagger generation

### PR DESCRIPTION
**Pull request by defong**
### Context
Genocide against swagger generation

### Changes proposed in this pull request
Committed genocide against swagger generation
nuget package now also includes 

- ManageCourses.ApiClient.dll
- ManageCourses.Api.dll
- ManageCourses.Domain.dll

### Guidance to review
Ignore the commented out stuff
Its only functioning as per the smoke test but smoke test does not cover all

The nuget package
https://www.nuget.org/packages/GovUk.Education.ManageCourses.ApiClient/0.22.0-death-to-swagger-generation.31

### CODE REVIEW

File: src/ManageCourses.ApiClient/HttpClientWrapper.cs [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+            this.wrapped = wrapped;
+        }
+
+        public async Task<HttpResponseMessage> GetAsync(Uri queryUri)
+        {
+            var msg = $"API GET Failed uri {queryUri}";
+            try
+            {
+                var result = await wrapped.GetAsync(queryUri);
+
```
Comment (MartinBentley) [inline src/ManageCourses.ApiClient/HttpClientWrapper.cs]: result.EnsureSuccessStatusCode ?

Comment (MartinBentley) [inline src/ManageCourses.ApiClient/HttpClientWrapper.cs]: Can you not just use result.EnsureSuccessStatusCode() to throw an exception for you?

Comment (defong) [reply src/ManageCourses.ApiClient/HttpClientWrapper.cs]: Nope, we need to fish out the status code to throw the exception

File: tests/ManageCourses.Tests/UnitTesting/Client/ManageCoursesApiClientTest.cs [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+using System.Web;
+using Newtonsoft.Json;
+
+namespace GovUk.Education.ManageCourses.Tests.UnitTesting.Client
+{
+    [TestFixture]
+    public class ManageCoursesApiClientTest
+    {
+
+        private ManageCoursesApiClient sut;
```
Comment (timabell) [inline tests/ManageCourses.Tests/UnitTesting/Client/ManageCoursesApiClientTest.cs]: call it what it is. `manageCoursesApiClient`

File: tests/ManageCourses.Tests/UnitTesting/Client/ManageCoursesApiClientTest.cs [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+             sut.AcceptTerms_IndexAsync().Wait();
+             mockHttp.VerifyAll();
+        }
+
+        [Test]
+        public void AccessRequest_IndexAsync()
+        {
+            var request  = new Api.Model.AccessRequest();
+            var controller = "accessrequest";
+            var leaf = "";
```
Comment (timabell) [inline tests/ManageCourses.Tests/UnitTesting/Client/ManageCoursesApiClientTest.cs]: inline unchanged var

File: tests/ManageCourses.Tests/UnitTesting/Client/ManageCoursesApiClientTest.cs [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+            result.Should().AllBeAssignableTo<IEnumerable<UserOrganisation>>();
+            mockHttp.VerifyAll();
+        }
+
+        public void Publish_PublishCourseToSearchAndCompareAsync()
+        {
+            var instCode = "instCode";
+            var courseCode = "courseCode";
+            var controller = "publish";
+            var leaf = $"/course/{instCode}/{courseCode}";
```
Comment (timabell) [inline tests/ManageCourses.Tests/UnitTesting/Client/ManageCoursesApiClientTest.cs]: I'd inline all the leaf vars in the file

File: tests/ManageCourses.Tests/UnitTesting/Client/ManageCoursesApiClientTest.cs [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+namespace GovUk.Education.ManageCourses.Tests.UnitTesting.Client
+{
+    [TestFixture]
+    public class ManageCoursesApiClientTest
+    {
+
+        private ManageCoursesApiClient sut;
+        private Mock<HttpClientWrapper> mockHttp;
+        private string accessToken = "accessToken";
+        private string baseurl = "http://baseurl.com";
```
[the remark this answers is not in the document]
Comment (defong) [reply tests/ManageCourses.Tests/UnitTesting/Client/ManageCoursesApiClientTest.cs]: then what well formed url i can use then?

File: tests/ManageCourses.Tests/UnitTesting/Client/ManageCoursesApiClientTest.cs (hunk 3 above)
Comment (defong) [reply tests/ManageCourses.Tests/UnitTesting/Client/ManageCoursesApiClientTest.cs]: superfluous

File: tests/ManageCourses.Tests/UnitTesting/Client/ManageCoursesApiClientTest.cs (hunk 4 above)
Comment (defong) [reply tests/ManageCourses.Tests/UnitTesting/Client/ManageCoursesApiClientTest.cs]: superfluous

try building the same url thats is either built by convention or via route or via httpget or via httppost, twice over

Then you recognise the need to use a template like `$"{baseurl}/api/{controller}{leaf}"`

File: tests/ManageCourses.Tests/UnitTesting/Client/ManageCoursesApiClientTest.cs (hunk 5 above)
[the remark this answers is not in the document]
Comment (defong) [reply tests/ManageCourses.Tests/UnitTesting/Client/ManageCoursesApiClientTest.cs]: Its not outgoing any way

File: src/ManageCourses.Api/Controllers/OrganisationsController.cs [hunk 6]
```diff
@@ -41,28 +41,28 @@ public ActionResult Get(string instCode)
...
+        //     var organisation = _dataService.GetUcasInstitutionForUser(name, instCode);
+        //     if (organisation == null)
+        //     {
+        //         return NotFound();
+        //     }
+        //     else
+        //     {
+        //         return Ok(organisation);
+        //     }
+        // }
```
[the remark this answers is not in the document]
Comment (defong) [reply src/ManageCourses.Api/Controllers/OrganisationsController.cs]: done

File: src/ManageCourses.ApiClient/HttpClientWrapper.cs [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+            {
+                throw;
+            }
+            catch(Exception ex)
+            {
+                throw new ManageCoursesApiException(msg, ex);
+            }
+        }
+
+        private async Task Guard(HttpResponseMessage result, string msg)
```
[the remark this answers is not in the document]
Comment (defong) [reply src/ManageCourses.ApiClient/HttpClientWrapper.cs]: done

File: tests/ManageCourses.Tests/UnitTesting/Client/ManageCoursesApiClientTest.cs (hunk 5 above)
[the remark this answers is not in the document]
Comment (defong) [reply tests/ManageCourses.Tests/UnitTesting/Client/ManageCoursesApiClientTest.cs]: done

File: tests/ManageCourses.Tests/UnitTesting/Client/ManageCoursesApiClientTest.cs [hunk 8]
```diff
@@ -0,0 +1,190 @@
...
+
+        [SetUp]
+        public void SetUp()
+        {
+            mockHttp = new Mock<HttpClientWrapper>(MockBehavior.Strict, new object[1]);
+            // mockHttp.Setup(x => x.GetAccessToken()).Returns(accessToken).Verifiable("Did not call GetAccessToken");
```
[the remark this answers is not in the document]
Comment (defong) [reply tests/ManageCourses.Tests/UnitTesting/Client/ManageCoursesApiClientTest.cs]: done

File: tests/ManageCourses.Tests/UnitTesting/Client/ManageCoursesApiClientTest.cs [hunk 9]
```diff
@@ -0,0 +1,190 @@
...
+            SetupPostUrlVerification($"{baseurl}/api/{controller}{leaf}");
+
+            // var url = $"{baseurl}/api/{controller}{leaf}";
+            // var expectedurl = It.Is<Uri>(y => y.AbsoluteUri == url);
+
+            //  mockHttp.Setup(x => x.PostAsync(It.Is<Uri>(y => y.AbsoluteUri == url), It.IsAny<System.Net.Http.StringContent>())).ReturnsAsync(
+            //     new System.Net.Http.HttpResponseMessage() {
+            //         StatusCode = HttpStatusCode.OK
+            //     }
+            // ).Verifiable();
```
[the remark this answers is not in the document]
Comment (defong) [reply tests/ManageCourses.Tests/UnitTesting/Client/ManageCoursesApiClientTest.cs]: done

File: tests/ManageCourses.Tests/UnitTesting/Client/ManageCoursesApiClientTest.cs (hunk 2 above)
Comment (defong) [reply tests/ManageCourses.Tests/UnitTesting/Client/ManageCoursesApiClientTest.cs]: done, just for you :)

**Comment by defong** [comment]
@timabell have another go

**Pull request merged**